### PR TITLE
Nodejs week2

### PR DESCRIPTION
**Pull request by oyelajakenny**
Added two categories of routes: meals and reservations.

The routes for meals will go into /api/src/routers/meals.js
The reservation routes will live in /api/src/routers/reservations.js

### CODE REVIEW

File: api/src/routers/meals.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+  } catch (error) {
+    res.status(500).json({ error: "Unable to update meal" });
+  }
+});
+mealsRouter.delete("/:id", async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    await knex("meal").where({ id: id }).del();
+    res.status(200).json({ message: "deleted successfully" });
+  } catch (error) {
```
Comment (urbanogilson) [inline api/src/routers/meals.js:54]: The same idea described in the `PUT` can be applied here to validate the id

File: api/src/routers/meals.js [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    res.status(500).json({ error: "Unable to delete meal" });
+  }
+});
+
+export default mealsRouter;
```
Comment (urbanogilson) [inline api/src/routers/meals.js:59]: The routes are not enabled in the `index.js`

File: api/src/routers/reservations.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+  try {
+    const id = req.params.id;
+    await knex("Reservation").where({ id: id }).del();
+    res.status(200).json({ message: "deleted successfully" });
+  } catch (error) {
+    res.status(500).json({ error: "Unable to delete reservation" });
+  }
+});
+
+export default reservationsRouter;
```
Comment (urbanogilson) [inline api/src/routers/reservations.js:57]: The routes are not enabled in the `index.js`

**Review by urbanogilson — COMMENTED** [review]
Good job!

File: api/src/routers/meals.js [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    res.status(201).json({ message: "created successfully" });
+  } catch (error) {
+    next(error);
+  }
+});
+
+mealsRouter.get("/meals/:id", async (req, res, next) => {
+  try {
+    const id = req.params.id;
+    const meals = await knex("meal").where({ id: id }).first();
```
Comment (urbanogilson) [inline api/src/routers/meals.js:25]:
```suggestion
    const meals = await knex("meal").where({ id }).first();
```

File: api/src/routers/meals.js [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+  } catch (error) {
+    next(res.status(500).json({ error: "Unable to retrieve meal" }));
+  }
+});
+
+mealsRouter.put("/:id", async (req, res) => {
+  try {
+    const id = req.params.id;
+    const data = req.body;
+    await knex("meal").where({ id: id }).update(data);
```
Comment (urbanogilson) [inline api/src/routers/meals.js:44]:
```suggestion
    await knex("meal").where({ id }).update(data);
```

File: api/src/routers/meals.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    if (!id) {
```
Comment (urbanogilson) [inline api/src/routers/meals.js:45]: It's better to valid before running the query, and after the query execution you can check if it was successful

**Comment by oyelajakenny** [comment]
Thank you for your feedback @urbanogilson, All recommendation has now been implemented

**Pull request merged**